script/DeployLeveragedEthMainnet.s.sol

```diff
@@ -1,73 +1,38 @@
-import {AggregatorV3Interface} from "../src/interfaces/chainlink/AggregatorV3Interface.sol";
-import {IVault} from "../src/interfaces/balancer/IVault.sol";
-import {ISwapRouter} from "../src/interfaces/uniswap/ISwapRouter.sol";
+import {DeployLeveragedEth} from "./base/DeployLeveragedEth.sol";
 import {scWETH} from "../src/steth/scWETH.sol";
 import {scUSDC} from "../src/steth/scUSDC.sol";
 import {sc4626} from "../src/sc4626.sol";
-import {MainnetDeployBase} from "./base/MainnetDeployBase.sol";
+import {Constants as C} from "../src/lib/Constants.sol";
+import {ISwapRouter} from "../src/interfaces/uniswap/ISwapRouter.sol";
 
-contract DeployScript is MainnetDeployBase {
-    function run() external returns (scWETH scWeth, scUSDC scUsdc) {
-        vm.startBroadcast(deployerPrivateKey);
+contract DeployScript is DeployLeveragedEth {
+    function run() external {
+        _deploy();
 
-        scWETH.ConstructorParams memory scWethParams = scWETH.ConstructorParams({
-            admin: deployerAddress,
-            keeper: keeper,
-            targetLtv: 0.7e18,
-            slippageTolerance: 0.99e18,
-            aavePool: IPool(C.AAVE_V3_POOL),
-            aaveAwstEth: IAToken(C.AAVE_V3_AWSTETH_TOKEN),
-            aaveVarDWeth: ERC20(C.AAVE_V3_VAR_DEBT_WETH_TOKEN),
-            curveEthStEthPool: ICurvePool(C.CURVE_ETH_STETH_POOL),
-            stEth: ILido(C.STETH),
-            wstEth: IwstETH(C.WSTETH),
-            weth: WETH(payable(C.WETH)),
-            stEthToEthPriceFeed: AggregatorV3Interface(C.CHAINLINK_STETH_ETH_PRICE_FEED),
-            balancerVault: IVault(C.BALANCER_VAULT)
-        });
+        _postDeployment();
+    }
 
-        scWeth = new scWETH(scWethParams);
+    function _postDeployment() internal {
+        vm.startBroadcast(deployerPrivateKey);
 
+        // scWETH
         weth.deposit{value: 0.01 ether}(); // wrap 0.01 ETH into WETH
+        console2.log("eth deposited for weth");
         _deposit(scWeth, 0.01 ether); // 0.01 WETH
+        console2.log("weth deposited into wethContract");
 
-        _transferAdminRoleToMultisig(scWeth, deployerAddress);
-
-        scUSDC.ConstructorParams memory scUsdcParams = scUSDC.ConstructorParams({
-            admin: deployerAddress,
-            keeper: keeper,
-            scWETH: scWeth,
-            usdc: usdc,
-            weth: WETH(payable(C.WETH)),
-            aavePool: IPool(C.AAVE_V3_POOL),
-            aavePoolDataProvider: IPoolDataProvider(C.AAVE_V3_POOL_DATA_PROVIDER),
-            aaveAUsdc: IAToken(C.AAVE_V3_AUSDC_TOKEN),
-            aaveVarDWeth: ERC20(C.AAVE_V3_VAR_DEBT_WETH_TOKEN),
-            uniswapSwapRouter: ISwapRouter(C.UNISWAP_V3_SWAP_ROUTER),
-            chainlinkUsdcToEthPriceFeed: AggregatorV3Interface(C.CHAINLINK_USDC_ETH_PRICE_FEED),
-            balancerVault: IVault(C.BALANCER_VAULT)
-        });
-
-        scUsdc = new scUSDC(scUsdcParams);
-
+        // scUSDC
+        weth.deposit{value: 0.01 ether}();
```

Comment:
You are doing weth.deposit twice. Once here and then again inside the _swapWethForUsdc method. Should delete this line.

Reply:
also on line 24 yeah?

Reply:
works with line 24 removed